src/main/java/com/dreammaster/gthandler/ModIDs.java

```diff
@@ -0,0 +1,26 @@
+package com.dreammaster.gthandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import cpw.mods.fml.common.Loader;
+
+public enum ModIDs {
+
+    EnderIO("EnderIO"),
+    TinkerConstruct("TConstruct");
+
+    public final String modID;
+
+    private final Map<String, Boolean> modsLoaded = new HashMap<>();
```

Comment:
No need for a map here, just make it a `private Boolean modLoaded = null;`, there's one per instance. If a map was needed, it should probably be static in that case (and maybe concurrent)

Reply:
i removed the map. Is it ok like that? I just don't know when the enum constructor will be evaluated at runtime, so dunno if what i did in the constructor will work or not.

Reply:
All constructors run during class loading, so I'd probably make it a Boolean and change it from null to a value on first use instead

Reply:
should be good now